ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -185,6 +188,25 @@ def start_trainer(self, trainer: Trainer, env_manager: EnvManager) -> None:
             trainer.write_tensorboard_text("Hyperparameters", trainer.parameters)
         env_manager.set_policy(trainer.brain_name, trainer.policy)
 
+    def _create_trainer_and_manager(
+        self, env_manager: EnvManager, brain_name: str
+    ) -> None:
+        trainer = self.trainer_factory.generate(env_manager.external_brains[brain_name])
+        self.start_trainer(trainer, env_manager)
+        agent_manager = AgentManager(
+            processor=AgentProcessor(
+                trainer.policy,
+                trainer.stats_reporter,
+                trainer.parameters.get("time_horizon", sys.maxsize),
+            ),
+            trajectory_queue=Queue(),
+            policy_queue=Queue(),
+        )
+        agent_manager.processor.publish_trajectory_queue(agent_manager.trajectory_queue)
```

Comment:
This feels strange that the AgentManager doesn't subscribe to its own trajectory_queue. I'd recommend making AgentManager its own class and set this up in `AgentManager.__init__` (you can maybe do this as a namedtuple still, but I think you have to do it in `__new__` instead)

Reply:
I made AgentManager a subclass of AgentProcessor, that contains 2 queues. Didn't integrate it with AgentProcessor since in the future we might have processors that subscribe to multiple queues.